src/test/java/be/syntra/devshop/DevshopFront/services/CartServiceImplTest.java

```diff
@@ -102,4 +136,58 @@ void willRemoveProductIfNumberInCartIsZeroWhenRemovingOneTest() {
         // then
         assertEquals(currentCart.getProducts().size(), 0);
     }
+
+    @Test
+    void payCartTest() {
+        //given
+        currentCart = CartUtils.getCartWithOneDummyProduct();
+        final String cartDtoAsJson = jsonUtils.asJsonString(currentCart);
+        final String expectedEndpoint = baseUrl + endpoint;
+        currentCart.setUser(principal.getName());
+
+        mockServer
+                .expect(requestTo(expectedEndpoint))
+                .andExpect(method(HttpMethod.POST))
+                .andRespond(
+                        MockRestResponseCreators
+                                .withStatus(HttpStatus.CREATED)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .body(cartDtoAsJson));
+
+        // when
+        StatusNotification statusNotification = cartService.payCart(principal.getName());
+
+        // then
+        mockServer.verify();
+        assertEquals(StatusNotification.SUCCESS, statusNotification);
+    }
+
+    @Test
+    void getTotalCartPriceTest() {
```

Comment:
This is a flawed test. You're setting the call to the service as the field for the Dto and asserting it's exact the same thing as the call to the service. 
To fix -> assign value to another variable and set the variable to be the dto's value (or inline it)
Add the // when, and there you're going to set the result of the service's call to another variable. Assert THAT result to the dto's value instead of calling the service again